src/MensaWidget/MWFrame.class.st

```diff
@@ -40,94 +41,105 @@ MWFrame class >> contentExtent [
 ]
 
 {
-	#category : #'as yet unclassified',
-	#'squeak_changestamp' : 'Oscar 6/16/2024 20:44'
+	#category : #constants,
+	#'squeak_changestamp' : 'Louis Christ 6/18/2024 19:24'
 }
-MWFrame class >> contentPosition [
+MWFrame class >> scrollBarThickness [
 
-	^ 5 @ 30
+	^ 14 * RealEstateAgent scaleFactor 
 ]
 
 {
 	#category : #constants,
-	#'squeak_changestamp' : 'Oscar 6/5/2024 01:01'
+	#'squeak_changestamp' : 'Louis Christ 6/18/2024 14:45'
 }
-MWFrame class >> scrollBarThickness [
+MWFrame class >> windowName [
 
-	^ 20
+	^ 'Mensa Widget'
 ]
 
 {
 	#category : #dayManipulation,
-	#'squeak_changestamp' : 'Oscar 6/16/2024 23:56'
+	#'squeak_changestamp' : 'Louis Christ 6/18/2024 15:34'
 }
-MWFrame >> addMeals: aDate [
-
-	| contentMorph |
+MWFrame >> abandonContent [
 	
-	contentMorph := self buildContentMorph.
-	(MWDataGettingWrapper getMealsOnDate: aDate yyyymmdd) do: 
-		[:each | contentMorph addMorph: each getCanteenMorph].
-	self contentMorph scroller 
-		layoutPolicy: TableLayout new;
-		addMorph: contentMorph
+	self contentMorph submorphs do: [:each | each abandon]
 
 	
 ]
 
 {
 	#category : #creation,
-	#'squeak_changestamp' : 'Oscar 6/16/2024 23:52'
+	#'squeak_changestamp' : 'Louis Christ 6/18/2024 17:40'
 }
-MWFrame >> buildContentMorph [
+MWFrame >> addContentMorph [
 
-	^ AlignmentMorph newColumn 
+	self contentMorph: (AlignmentMorph newColumn 
 		 cellInset: self class cellSpacing;
 		 extent: self class contentExtent;
-		 hResizing: #spaceFill;
-		 vResizing: #spaceFill
+		vResizing: #shrinkWrap;
+		yourself).
+	self scroller addMorph: self contentMorph
 ]
 
 {
-	#category : #creation,
-	#'squeak_changestamp' : 'Oscar 6/16/2024 23:51'
+	#category : #dayManipulation,
+	#'squeak_changestamp' : 'Louis Christ 6/18/2024 16:05'
 }
-MWFrame >> buildLayout [
+MWFrame >> addMeals: aDate forTime: aTime [
```

Comment:
i think a name like `addMealsOn: aDate forTime: aTime` would be a bit more fitting